### PR TITLE
Examples align

### PR DESCRIPTION
**Pull request by ArekBalysNordic**

### CODE REVIEW

File: examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (kkasperczyk-no) [inline examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay:2]: The copyright header is not correct here and in several other places. It should be the one used by the Project CHIP authors.

**Review by kkasperczyk-no — COMMENTED** [review]
Shouldn't we also add Wi-Fi support to the lighting-app?

File: examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay (hunk 1 above)
Comment (ArekBalysNordic) [reply examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay:2]: > Shouldn't we also add Wi-Fi support to the lighting-app?

I guess, not yet, we align to the NCS 2.1.1.

Comment (kkasperczyk-no) [reply examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay:2]: Yes and no :) We align to the NCS 2.1.1, but next upmerge will be done from Matter upstream to NCS main, that has lighting-app WiFi support. So in such case there will be problem with upmerge, because samples will not be aligned. I mean it's ok for me to not add it now, but I think functionality should not conflict with NCS 2.1.1 code, right?

Comment (ArekBalysNordic) [reply examples/all-clusters-app/nrfconnect/boards/nrf7002dk_nrf5340_cpuapp.overlay:2]: hmm, ok agree, it can speed up next upmerge. I will add it too.

File: docs/guides/nrfconnect_android_commissioning.md [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
-This guide references the nRF52840 DK and the door lock example application
-based on the nRF Connect platform, but the instructions are also valid for the
-nRF Connect lighting example application and can be adapted to other platforms
-and applications as well.
+This guide references the nRF52840 DK and Matter nRF Connect Lighting Example
+Application that communicates with other nodes over a Thread network, but the
+instructions can be adapted to other platforms and applications. In particular,
+some sections of this guide include deviations from the original procedure that
+are needed to test a Wi-Fi device.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_android_commissioning.md:11]:
```suggestion
instructions can be adapted to other platforms and applications. For instance,
some sections of this guide include steps for testing a Wi-Fi device, which are adapted from the original Thread-based procedure.
```

File: docs/guides/nrfconnect_android_commissioning.md [hunk 3]
```diff
@@ -60,31 +61,42 @@ applications:
 You need the following hardware and software for commissioning the nRF Connect
 accessory using Android CHIPTool:
 
--   Two nRF52840 DK (PCA10056)
+-   1x smartphone with Android 8+
+-   1x Wi-Fi Access Point supporting IPv6 (without the IPv6 Router Advertisement
```
Comment (greg-fer) [inline docs/guides/nrfconnect_android_commissioning.md:65]: This is needed for both Thread and Wi-Fi?

File: docs/guides/nrfconnect_android_commissioning.md [hunk 4]
```diff
@@ -60,31 +61,42 @@ applications:
...
+    Guard enabled on the router)
+-   1x nRF52840 DK (PCA10056) for running the example application. You can
+    replace this DK with another compatible device, such as the nRF5340 DK or
+    nRF7002 DK. nRF52840 DK and nRF5340 DK can be used to test Matter over
+    Thread, and nRF7002 DK can be used to test Matter over Wi-Fi.
+-   1x nRF52840 DK for running the
+    [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor)
+    firmware. You can replace this DK with another compatible device, such as
+    the nRF52840 Dongle.
```
Comment (greg-fer) [inline docs/guides/nrfconnect_android_commissioning.md:74]: Put the note from lines 76-77 before the piece of hardware needed:
```suggestion
- When testing a Thread device, you also need:

   -   1x nRF52840 DK for running the
        [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor)
        firmware. You can replace this DK with another compatible device, such as
        the nRF52840 Dongle.
   -   1x Raspberry Pi Model 3B+ or newer (along with an SD card with at least 8 GB
        of memory)
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 5]
```diff
@@ -9,13 +9,9 @@ creating your own application.
 
 The example is based on
 [Matter](https://github.com/project-chip/connectedhomeip) and Nordic
-Semiconductor's nRF Connect SDK, and was created to facilitate testing and
-certification of a Matter device communicating over a low-power, 802.15.4 Thread
-network.
+Semiconductor's nRF Connect SDK, and works as a Matter accessory device, meaning it can be paired and controlled remotely over a Matter network built on top of a low-power 802.15.4 Thread or Wi-Fi network.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:12]:
```suggestion
Semiconductor's nRF Connect SDK, and was created to facilitate testing and
certification of a Matter device communicating over a low-power, 802.15.4 network, either Thread or Wi-Fi.
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 6]
```diff
@@ -9,13 +9,9 @@ creating your own application.
...
 [Matter](https://github.com/project-chip/connectedhomeip) and Nordic
-Semiconductor's nRF Connect SDK, and was created to facilitate testing and
-certification of a Matter device communicating over a low-power, 802.15.4 Thread
-network.
+Semiconductor's nRF Connect SDK, and works as a Matter accessory device, meaning it can be paired and controlled remotely over a Matter network built on top of a low-power 802.15.4 Thread or Wi-Fi network. 
 
-The example behaves as a Matter accessory, that is a device that can be paired
-into an existing Matter network and can be controlled by this network. The
-device works as a Thread Minimal End Device.
+Support for both Thread and Wi-Fi is mutually exclusive and depends on the hardware platform, so only one protocol can be supported for a specific lock device. In the case of Thread, this device works as a Thread Sleepy End Device. You can use this sample as a reference for creating your application.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:14]:
```suggestion
The example behaves as a Matter accessory, that is a device that can be paired
into an existing Matter network and can be controlled by this network. 

In the case of Thread, this device works as a Thread Sleepy End Device.
Support for both Thread and Wi-Fi is mutually exclusive and depends on the hardware platform, so only one protocol can be supported for a specific lock device.
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 7]
```diff
@@ -44,6 +40,13 @@ device works as a Thread Minimal End Device.
 
 <hr>
 
+## IPv6 network support
+
+The development kits for this sample offer the following IPv6 network support for Matter:
+
+* Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``, and ``nrf21540dk_nrf52840``.
+* Matter over Wi-Fi is supported for ``nrf5340dk_nrf5340_cpuapp`` with the ``nrf7002_ek`` shield attached or for ``nrf7002dk_nrf5340_cpuapp``.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:48]: Move this section under Requirements, on the same level as Supported devices (###). Also update the table of contents.

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 8]
```diff
@@ -75,14 +84,14 @@ commissioner role.
 
 To start the rendezvous, the controller must get the commissioning information
 from the Matter device. The data payload is encoded within a QR code, printed to
-the UART console.
-
-#### Thread provisioning
+the UART console, and shared using an NFC tag. NFC tag emulation starts
+automatically when Bluetooth LE advertising is started and stays enabled until
+Bluetooth LE advertising timeout expires.
+#### hread or Wi-Fi provisioning
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:90]: Add a blank line and fix typo:
```suggestion

#### Thread or Wi-Fi provisioning
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 9]
```diff
@@ -75,14 +84,14 @@ commissioner role.
...
 
 Last part of the rendezvous procedure, the provisioning operation involves
-sending the Thread network credentials from the Matter controller to the Matter
-device. As a result, device is able to join the Thread network and communicate
-with other Thread devices in the network.
+sending the Thread / Wi-Fi network credentials from the Matter controller to the Matter device. As a result, the device joins the Thread or Wi-Fi network and can
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:93]:
```suggestion
sending the Thread or Wi-Fi network credentials from the Matter controller to the Matter device. As a result, the device joins the Thread or Wi-Fi network and can
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 10]
```diff
@@ -105,6 +114,8 @@ The example supports building and running on the following devices:
 | [nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK)         | `nrf52840dk_nrf52840`      | <details><summary>nRF52840 DK</summary><img src="../../platform/nrfconnect/doc/images/nRF52840_DK_info-medium.jpg" alt="nRF52840 DK"/></details>        |
 | [nRF5340 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF5340-DK)           | `nrf5340dk_nrf5340_cpuapp` | <details><summary>nRF5340 DK</summary><img src="../../platform/nrfconnect/doc/images/nRF5340_DK_info-medium.jpg" alt="nRF5340 DK"/></details>           |
 | [nRF52840 Dongle](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-Dongle) | `nrf52840dongle_nrf52840`  | <details><summary>nRF52840 Dongle</summary><img src="../../platform/nrfconnect/doc/images/nRF52840_Dongle-medium.jpg" alt="nRF52840 Dongle"/></details> |
+| [nRF7002 DK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf7002.html#nrf7002dk-nrf5340)   | `nrf7002dk_nrf5340_cpuapp` | <details><summary>nRF7002DK</summary><img src="../../platform/nrfconnect/doc/images/nrf7002dk.jpg" alt="nRF7002 DK"/></details>    |
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:117]: I don't see any mention of the ``nrf7002_ek`` or ``nrf21540dk_nrf52840`` here, yet you mention them in the IPv6 network support section...

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 11]
```diff
@@ -55,17 +58,23 @@ and [Zephyr RTOS](https://zephyrproject.org/). Visit Matter's
...
-has Thread disabled, and it should be paired with Matter controller and get
-configuration from it. Some actions required before establishing full
-communication are described below.
+The sample uses buttons for changing the lock and device states, and LEDs to show the state of these changes. You can test it in the following ways:
+
+- Standalone, using a single DK that runs the door lock application.
+
+- Remotely over the Thread or the Wi-Fi protocol, which in either case requires more devices, including a Matter controller that you can configure either on a PC or a mobile device.
+
+By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:67]: Move this whole line to line 58, before the list about standalone/remote testing.

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 12]
```diff
@@ -137,21 +148,38 @@ following states are possible:
     Bluetooth LE.
 
 -   _Short Flash Off (950ms on/50ms off)_ &mdash; The device is fully
-    provisioned, but does not yet have full Thread network or service
+    provisioned but does not yet have full Thread network or service
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:151]:
```suggestion
    provisioned, but does not yet have full Thread network or service
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 13]
```diff
@@ -137,21 +148,38 @@ following states are possible:
...
+-   _Solid On_ &mdash; The device is fully provisioned.
 
 **Button 1** can be used for the following purposes:
 
--   _Pressed for 6 s_ &mdash; Initiates the factory reset of the device.
-    Releasing the button within the 6-second window cancels the factory reset
-    procedure. **LEDs 1-4** blink in unison when the factory reset procedure is
-    initiated.
+-   _Pressed for less than 3 s_ &mdash; Initiates the OTA software update
+    process. This feature is disabled by default but can be enabled by
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:159]:
```suggestion
    process. This feature is disabled by default, but can be enabled by
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 14]
```diff
@@ -137,21 +148,38 @@ following states are possible:
...
+    following the
+    [Building with Device Firmware Upgrade support](#building-with-device-firmware-upgrade-support)
+    instructions.
+
+-   _Pressed for more than 3 s_ &mdash; initiates the factory reset of the     device.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:164]:
```suggestion
-   _Pressed for more than 3 s_ &mdash; initiates the factory reset of the device. 
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md (hunk 11 above)
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:67]: Also about this line: we are adding information about manual discovery, but there is no mention of this requirement in the Testing section or elsewhere in the document, and so even the UI instructions are confusing. I'd suggest adding the relevant information to the Testing section.

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by greg-fer — COMMENTED** [review]
@ArekBalysNordic , reviewed only one example, but please apply edits to all of them, as the changes are similar.

File: docs/guides/nrfconnect_android_commissioning.md (hunk 3 above)
Comment (ArekBalysNordic) [reply docs/guides/nrfconnect_android_commissioning.md:65]: Yes both of them. Raspberry Pi as Thread Border Router uses this Wi-Fi access point.

File: .github/workflows/examples-nrfconnect.yaml [hunk 15]
```diff
@@ -192,6 +192,38 @@ jobs:
                     nrfconnect nrf5340dk_nrf5340_cpuapp lighting-app \
                     examples/lighting-app/nrfconnect/build/zephyr/zephyr.elf \
                     /tmp/bloat_reports/
+            - name: Build example nRF Connect SDK Lock App on nRF7002 PDK
```
Comment (Damian-Nordic) [inline .github/workflows/examples-nrfconnect.yaml:195]: Could we build only all clusters app on nRF70 on PR and the rest only if `if: github.event_name == 'push' || steps.changed_paths.outputs.nrfconnect == 'true'`? Otherwise, it will take lots of time to build all the examples.

File: docs/guides/nrfconnect_android_commissioning.md [hunk 16]
```diff
@@ -68,20 +67,18 @@ accessory using Android CHIPTool:
...
 
+-   1x nRF52840 DK for running the
+    [OpenThread Radio Co-Processor](https://openthread.io/platforms/co-processor)
+    firmware. You can replace this DK with another compatible device, such as
+    the nRF52840 Dongle.
+    
 -   1x Raspberry Pi Model 3B+ or newer (along with an SD card with at least 8 GB
     of memory)
 
-    > _Note:_ This piece of hardware is only needed if you're testing a Thread
```
Comment (Damian-Nordic) [inline docs/guides/nrfconnect_android_commissioning.md:82]: Why is this removed?

File: examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp [hunk 17]
```diff
@@ -40,24 +40,28 @@
 #include <logging/log.h>
 #include <zephyr.h>
 
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
```
Comment (Damian-Nordic) [inline examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp:43]: I think we shouldn't use `CONFIG_MATTER_LOG_LEVEL` to control log level of the application. It is for controlling Matter library.

File: examples/pump-app/nrfconnect/main/AppTask.cpp [hunk 18]
```diff
@@ -45,45 +46,54 @@
...
-#define BUTTON_RELEASE_EVENT 0
-
-#define PCC_CLUSTER_ENDPOINT 1
-#define ONOFF_CLUSTER_ENDPOINT 1
-
 namespace {
+constexpr uint32_t kFactoryResetTriggerTimeout      = 3000;
+constexpr uint32_t kFactoryResetCancelWindowTimeout = 3000;
+constexpr size_t kAppEventQueueSize                 = 10;
+constexpr EndpointId PccClusterEndpoint             = 1;
```
Comment (Damian-Nordic) [inline examples/pump-app/nrfconnect/main/AppTask.cpp:60]: We should prefix all constants with `k` for consistency.

**Review by kkasperczyk-no — COMMENTED** [review]
There is lack of pm_static_dfu file for nRF7002DK in all-clusters-minimal-app

**Comment by kkasperczyk-no** [comment]
> There is lack of pm_static_dfu file for nRF7002DK in all-clusters-minimal-app

Nvm it doesn't support Wi-Fi, that's why :)

File: docs/guides/nrfconnect_android_commissioning.md (hunk 16 above)
Comment (ArekBalysNordic) [reply docs/guides/nrfconnect_android_commissioning.md:82]: Oh, Is it the note scope? I thought that it's unnecessarily duplicated.

File: examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp (hunk 17 above)
Comment (ArekBalysNordic) [reply examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp:43]: Hmm, so what should we use instead of that? As I remember it was added as a workaround because we used LOG notation from Zephyr in our samples and we wanted to have the same log level in application and Matter library.

Comment (Damian-Nordic) [reply examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp:43]: It's of course fine to use LOG_ macros for the application but what is not right is using Matter's log level to configure the application's log level. It's like BLE samples using log level of one of BLE host or BLE controller to configure their log level. It's just unnecessary coupling. We can pass harcoded `LOG_LEVEL_DBG` (like some other samples in sdk-nrf do) or we can define another Kconfig in each sample to configure the log level. I once needed to reduce the flash size of Matter samples so I disabled Matter logs, and was very surprised that the application logs disappeared as well :).

File: examples/all-clusters-minimal-app/nrfconnect/README.md (hunk 11 above)
Comment (ArekBalysNordic) [reply examples/all-clusters-minimal-app/nrfconnect/README.md]: You can find it in the `Bluetooth LE advertising` section. Maybe it's worth to add a link to that section in this sentence.

File: examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp (hunk 17 above)
Comment (ArekBalysNordic) [reply examples/all-clusters-minimal-app/nrfconnect/main/AppTask.cpp:43]: Ok, I see the point :) I've not thought about it, seen in NCS samples `LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);`, and just copied this line. I agree with you completely, and will add Kconfig to control the application log level.

File: examples/all-clusters-app/nrfconnect/README.md [hunk 19]
```diff
@@ -75,14 +78,14 @@ commissioner role.
 
 To start the rendezvous, the controller must get the commissioning information
 from the Matter device. The data payload is encoded within a QR code, printed to
-the UART console.
+the UART console, and shared using an NFC tag. NFC tag emulation starts
```
Comment (greg-fer) [inline examples/all-clusters-app/nrfconnect/README.md:81]:
```suggestion
the UART console, and shared using an NFC tag. The emulation of the NFC tag starts
```

File: examples/all-clusters-app/nrfconnect/README.md [hunk 20]
```diff
@@ -105,9 +108,19 @@ The example supports building and running on the following devices:
...
+| [nRF7002 DK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf7002.html#nrf7002dk-nrf5340)   | `nrf7002dk_nrf5340_cpuapp` | <details><summary>nRF7002DK</summary><img src="../../platform/nrfconnect/doc/images/nrf7002dk.jpg" alt="nRF7002 DK"/></details>    |
+
 
 <hr>
 
+### IPv6 network support
+
+The development kits for this sample offer the following IPv6 network support for Matter:
+
+* Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``.
```
Comment (greg-fer) [inline examples/all-clusters-app/nrfconnect/README.md:120]:
```suggestion
* Matter over Thread is supported for ``nrf52840dk_nrf52840`` and ``nrf5340dk_nrf5340_cpuapp``.
```

File: examples/all-clusters-app/nrfconnect/README.md [hunk 21]
```diff
@@ -137,21 +150,37 @@ following states are possible:
     Bluetooth LE.
 
 -   _Short Flash Off (950ms on/50ms off)_ &mdash; The device is fully
-    provisioned, but does not yet have full Thread network or service
-    connectivity.
+    provisioned, but does not yet have full Thread network, Wi-Fi network or service connectivity.
```
Comment (greg-fer) [inline examples/all-clusters-app/nrfconnect/README.md:153]:
```suggestion
    provisioned, but does not yet have full connectivity for Thread or Wi-Fi network, or the related services.
```

File: examples/all-clusters-minimal-app/nrfconnect/README.md [hunk 22]
```diff
@@ -55,11 +55,9 @@ and [Zephyr RTOS](https://zephyrproject.org/). Visit Matter's
 [nRF Connect platform overview](../../../docs/guides/nrfconnect_platform_overview.md)
 to read more about the platform structure and dependencies.
 
-The Matter device that runs the all clusters application is controlled by the
-Matter controller device over the Thread protocol. By default, the Matter device
-has Thread disabled, and it should be paired with Matter controller and get
-configuration from it. Some actions required before establishing full
-communication are described below.
+By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do it. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```
Comment (greg-fer) [inline examples/all-clusters-minimal-app/nrfconnect/README.md:58]:
```suggestion
By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do this. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```

File: examples/light-switch-app/nrfconnect/README.md [hunk 23]
```diff
@@ -123,12 +144,10 @@ the UART console, and shared using an NFC tag. NFC tag emulation starts
...
 Bluetooth LE advertising timeout expires.
 
-#### Thread provisioning
+#### Thread or Wi-Fi provisioning
 
-Last part of the rendezvous procedure, the provisioning operation involves
-sending the Thread network credentials from the Matter controller to the Matter
-device. As a result, the device is able to join the Thread network and
-communicate with other Thread devices in the network.
+The Last part of the rendezvous procedure, the provisioning operation involves
```
Comment (greg-fer) [inline examples/light-switch-app/nrfconnect/README.md:149]:
```suggestion
The provisioning operation, which is the Last part of the rendezvous procedure, involves
```

File: examples/light-switch-app/nrfconnect/README.md [hunk 24]
```diff
@@ -207,6 +226,14 @@ The example supports building and running on the following devices:
...
 | [nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK) | `nrf52840dk_nrf52840`      | <details><summary>nRF52840 DK</summary><img src="../../platform/nrfconnect/doc/images/nRF52840_DK_info-medium.jpg" alt="nRF52840 DK"/></details> |
 | [nRF5340 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF5340-DK)   | `nrf5340dk_nrf5340_cpuapp` | <details><summary>nRF5340 DK</summary><img src="../../platform/nrfconnect/doc/images/nRF5340_DK_info-medium.jpg" alt="nRF5340 DK"/></details>    |
+| [nRF7002 DK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf7002.html#nrf7002dk-nrf5340)   | `nrf7002dk_nrf5340_cpuapp` | <details><summary>nRF7002DK</summary><img src="../../platform/nrfconnect/doc/images/nrf7002dk.jpg" alt="nRF7002 DK"/></details>    |
+
+### IPv6 network support
+
+The development kits for this sample offer the following IPv6 network support for Matter:
+
+* Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``.
```
Comment (greg-fer) [inline examples/light-switch-app/nrfconnect/README.md:235]:
```suggestion
* Matter over Thread is supported for ``nrf52840dk_nrf52840`` and ``nrf5340dk_nrf5340_cpuapp``.
```

File: examples/light-switch-app/nrfconnect/README.md [hunk 25]
```diff
@@ -62,6 +69,16 @@ and [Zephyr RTOS](https://zephyrproject.org/). Visit Matter's
 [nRF Connect platform overview](../../../docs/guides/nrfconnect_platform_overview.md)
 to read more about the platform structure and dependencies.
 
+By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do it. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```
Comment (greg-fer) [inline examples/light-switch-app/nrfconnect/README.md:72]:
```suggestion
By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do this. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```

File: examples/all-clusters-app/nrfconnect/README.md [hunk 26]
```diff
@@ -55,17 +56,19 @@ and [Zephyr RTOS](https://zephyrproject.org/). Visit Matter's
 [nRF Connect platform overview](../../../docs/guides/nrfconnect_platform_overview.md)
 to read more about the platform structure and dependencies.
 
-The Matter device that runs the all clusters application is controlled by the
-Matter controller device over the Thread protocol. By default, the Matter device
-has Thread disabled, and it should be paired with Matter controller and get
-configuration from it. Some actions required before establishing full
-communication are described below.
+By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do it. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```
Comment (greg-fer) [inline examples/all-clusters-app/nrfconnect/README.md:59]:
```suggestion
By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. You have to make the device discoverable manually (for security reasons). See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do this. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```

File: examples/lighting-app/nrfconnect/README.md [hunk 27]
```diff
@@ -58,21 +60,25 @@ and [Zephyr RTOS](https://zephyrproject.org/). Visit Matter's
 [nRF Connect platform overview](../../../docs/guides/nrfconnect_platform_overview.md)
 to read more about the platform structure and dependencies.
 
-The Matter device that runs the lighting application is controlled by the Matter
-controller device over the Thread protocol. By default, the Matter device has
-Thread disabled, and it should be paired with Matter controller and get
-configuration from it. Some actions required before establishing full
-communication are described below.
+By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. The device starts advertising automatically and you can commission the device within 15 minutes. If the advertising time elapsed you can re-enable it using buttons. See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do it. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```
Comment (greg-fer) [inline examples/lighting-app/nrfconnect/README.md:63]:
```suggestion
By default, the Matter accessory device has IPv6 networking disabled. You must pair it with the Matter controller over Bluetooth® LE to get the configuration from the controller to use the device within a Thread or Wi-Fi network. The device starts advertising automatically and you can commission the device within 15 minutes. If the advertising time elapsed you can re-enable it using buttons. See [Bluetooth LE advertising](#bluetooth-le-advertising) to learn how to do this. The controller must get the commissioning information from the Matter accessory device and provision the device into the network.
```

File: examples/lighting-app/nrfconnect/README.md [hunk 28]
```diff
@@ -171,9 +175,18 @@ The example supports building and running on the following devices:
...
 | [nRF52840 Dongle](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-Dongle) | `nrf52840dongle_nrf52840`  | <details><summary>nRF52840 Dongle</summary><img src="../../platform/nrfconnect/doc/images/nRF52840_Dongle-medium.jpg" alt="nRF52840 Dongle"/></details> |
+| [nRF7002 DK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf7002.html#nrf7002dk-nrf5340)   | `nrf7002dk_nrf5340_cpuapp` | <details><summary>nRF7002DK</summary><img src="../../platform/nrfconnect/doc/images/nrf7002dk.jpg" alt="nRF7002 DK"/></details>    |
 
 <hr>
 
+### IPv6 network support
+
+The development kits for this sample offer the following IPv6 network support for Matter:
+
+* Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``.
```
Comment (greg-fer) [inline examples/lighting-app/nrfconnect/README.md:186]:
```suggestion
* Matter over Thread is supported for ``nrf52840dk_nrf52840`` and ``nrf5340dk_nrf5340_cpuapp``.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Please apply the nits from added comments globally. Thanks.

**Review by greg-fer — APPROVED** [review]
